### PR TITLE
Update styles for language toggle

### PR DESCRIPTION
**Pull request by marinaaisa**
Bug/issue #122676572, if applicable: 

## Summary

Update styles for language toggle:
- Semibold font-weight and underline to links of the language list in small viewports if page has multiple languages to select from
- Semibold font-weight in links for the language dropdown in large viewports
- Light and dark mode for border colors
- Nav-menu-label-margin should be 5px

## Dependencies

NA

## Testing

Steps:
1. Go to a page with only one language available and assert that the summary specs are correct
2. Go to a page with multiple languages available and assert that the summary specs are correct

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: src/components/DocumentationTopic/DocumentationNav/LanguageToggle.vue [hunk 1]
```diff
@@ -315,7 +316,7 @@ $nav-menu-toggle-label-margin: 6px;
     display: inline-block;
 
     &:not(:first-child) {
-      border-left: 1px solid dark-color(fill-gray-tertiary);
+      border-left: 1px solid var(--color-fill-gray-tertiary);
```
Comment (hqhhuang) [inline src/components/DocumentationTopic/DocumentationNav/LanguageToggle.vue:319]: I think we want the `dark-color` variant here? The current color contrast of the divider line looks too weak to me.
image

**Review by hqhhuang — CHANGES_REQUESTED** [review]
Seems like we should make the toggle component wider to accommodate the bold and underline language text:
image

**Comment by marinaaisa** [comment]
> Seems like we should make the toggle component wider to accommodate the bold and underline language text: image

Fixed: https://github.com/apple/swift-docc-render/pull/781/commits/ae283a117efd72fbbc6a3040d4a1d797445413d6

**Review by hqhhuang — COMMENTED** [review]
Everything else looks good, except can you check on the color of the dividing line in the language toggle?

File: src/components/DocumentationTopic/DocumentationNav/LanguageToggle.vue (hunk 1 above)
Comment (hqhhuang) [reply src/components/DocumentationTopic/DocumentationNav/LanguageToggle.vue:319]: Edit: I meant that we should pick a different color. `fill-gray-tertiary` is darker in dark mode than in light mode. I would expect the opposite.

Comment (marinaaisa) [reply src/components/DocumentationTopic/DocumentationNav/LanguageToggle.vue]: You are right! Fixed: https://github.com/apple/swift-docc-render/pull/781/commits/8fa580cb913eccfd534461129c32d39fb56432d2

Comment (marinaaisa) [reply src/components/DocumentationTopic/DocumentationNav/LanguageToggle.vue]: Screenshot 2024-03-14 at 22 55 18

**Comment by marinaaisa** [comment]
> Everything else looks good, except can you check on the color of the dividing line in the language toggle?

It's fixed now! https://github.com/apple/swift-docc-render/pull/781/commits/8fa580cb913eccfd534461129c32d39fb56432d2

**Review by SamLanier — APPROVED** [review]
Design approved.

**Comment by marinaaisa** [comment]
@swift-ci test

**Review by hqhhuang — APPROVED** [review]

**Comment by marinaaisa** [comment]
@swift-ci test

**Pull request merged**